### PR TITLE
Add azure job for windows and restructure azure config files

### LINKED ISSUE #455: Add windows to azure ci
* The project is not tested in ci with windows and it would catch some issues earlier
* Not sure it it had been tried and some issue made it not feasible
* The setup for haskell-ide-engine could be used as reference: https://github.com/haskell/haskell-ide-engine/blob/master/.azure/windows-stack.yml

### PR DESCRIPTION
**Pull request by jneira**
* Tests are disabled for now cause there are [three failing ones.](https://github.com/digital-asset/ghcide/issues/474) They have to be:
  * Fixed :smile: 
  * Marked as broken in test code
  * Avoided using `--pattern`
  * But it could be done in a follow-up
* I've used some azure elements not present in existing jobs (using Cache@2, using matrix startegy). They can be update in (another) follow-up if you consider it could a improvement over them
* I've added branches starting with `azure` in the trigger, cause i've found them useful to test azure configurations without worrying of trigger manual executions

Closes #455

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -197,3 +198,68 @@ jobs:
                --data "{\"text\":\"<!here> *FAILED* $(Agent.JobName): <https://dev.azure.com/digitalasset/ghcide/_build/results?buildId=$(Build.BuildId)|$MESSAGE>\n\"}" \
                $(Slack.URL)
         condition: and(failed(), eq(variables['Build.SourceBranchName'], 'master'))
+  - job: ghcide_stack_windows
```
Comment (cocreature) [inline azure-pipelines.yml]: Can we use a single matrix for all platforms? It seems a bit weird that the Windows test setup looks very different from the Linux test setup and also makes it hard to figure out what is different. The matrix setup definily looks nicer than the mess we have currently.

Comment (jneira) [reply azure-pipelines.yml]: well the bash script and cache handling for linux and windows are different and i am not sure if they can be integrated without too much if's
what about a matrix per os like in hie?

**Comment by jneira** [comment]
@cocreature i've used the hie layout for azure configuration, using templates and a job for each operating system.
I think that it will be more maintainable this way cause it is probable that macos will be added (with its own way to installing deps) and the scripts of each step would be either one-liners or up to three if blocks. Even steps itself could be different. 
If we want to reuse bash script, like the used to send a notificaction we can extract it and call from the azure config file.

**Comment by cocreature** [comment]
Needs rebasing on master.

**Comment by jneira** [comment]
I rebased it but the build has failed for two windows jobs, they dont seem to be related with the changes though. I am rerunning it in my azure pipeline: https://dev.azure.com/jneira/haskell-language-server/_build/results?buildId=726

**Comment by jneira** [comment]
I wonder if 
/azp run
would work for me

**Comment by jneira** [comment]
@cocreature my pipeline ran succesfully https://dev.azure.com/jneira/haskell-language-server/_build/results?buildId=726

**Pull request merged**